bin/pygrb/pycbc_pygrb_plot_coh_ifosnr

```diff
@@ -20,189 +20,242 @@
+
+    return data
+
+
+# Plot lines representing deviations based on non-central chi-square
+def plot_deviation(percentile, snr_grid, y, ax, style):
+    """Plot deviations based on non-central chi-square"""
+
+    # ncx2: non-central chi-squared; ppf: percent point function
+    #ax.plot(snr_grid, scipy.stats.ncx2.ppf(percentile, 2, y*y)**0.5, style)
+
+    # Using interpolation to work around "saturation" given by the
+    # original code line (commented out above)
+    y_vals = scipy.stats.ncx2.ppf(percentile, 2, y*y)**0.5
+    y_vals = numpy.unique(y_vals)
+    x_vals = snr_grid[0:len(y_vals)]
+    n_vals = int(len(y_vals)/2)
+    f = scipy.interpolate.interp1d(x_vals[0:n_vals], y_vals[0:n_vals],
+                                   bounds_error=False, fill_value="extrapolate")
+    ax.plot(snr_grid, f(snr_grid), style)
+
 
+# =============================================================================
+# Main script starts here
+# =============================================================================
 description = 'Produces coherent SNR versus single IFO SNR plots.'
 usage = __program__ + ' [--options]'
-opts = pygrb_plot_opts_parser(usage=usage, description=description, version=__version__)
 
-if opts.verbose:
-    level = logging.INFO
-else:
-    level = logging.WARNING
-logging.basicConfig(format="%(asctime)s:%(levelname)s : %(message)s",
-                    level=level)
+parser = ppu.pygrb_initialize_plot_parser(usage=usage, description=description,
+                                          version=__version__)
+parser.add_argument("-t", "--trig-file", action="store",
+                    default=None, required=True,
+                    help="The location of the trigger file")
+parser.add_argument("-z", "--zoom-in", default=False, action="store_true",
+                    help="Output file a zoomed in version of the plot.")
+opts = parser.parse_args()
 
-trig_file  = os.path.abspath(opts.trig_file)
-inj_file   = None
+init_logging(opts.verbose, format="%(asctime)s: %(levelname)s: %(message)s")
+
+# Check options
+trig_file = os.path.abspath(opts.trig_file)
+inj_file = None
 if opts.inj_file:
     inj_file = os.path.abspath(opts.inj_file)
-outfile       = opts.output_file
-zoomedoutfile = opts.zoomed_output_file
-if zoomedoutfile == None:
-    msg = "Please specify a pathname with the zoomed-ouptut-file option"
-    logging.error(msg)
-    sys.exit()
+zoom_in = opts.zoom_in
 veto_files = []
 if opts.veto_directory:
-    veto_string = ','.join([str(i) for i in range(2,opts.veto_category+1)])
+    veto_string = ','.join([str(i) for i in range(2, opts.veto_category+1)])
     veto_files = glob.glob(opts.veto_directory +'/*CAT[%s]*.xml' %(veto_string))
 ifo = opts.ifo
-if ifo == None:
-    msg = "Please specify an interferometer"
-    logging.error(msg)
-    sys.exit()
+if ifo is None:
+    err_msg = "Please specify an interferometer"
+    logging.error(err_msg)
+    raise RuntimeError(err_msg)
```

Comment:
No need for this check, just use `required=True` in `add_argument()`.

Reply:
Hmm after looking at the rest of the code, maybe what I suggested is not easy or convenient. Your choice, but at least
do the `parser.error(err_msg)` replacement as suggested above.